### PR TITLE
Fixed Anvil format.

### PR DESCRIPTION
**Pull request by IrishPacks**
Fixed Anvil format to have a chunk parameter as it lacked of one.

### CODE REVIEW

File: src/pocketmine/Server.php [hunk 1]
```diff
@@ -884,7 +884,7 @@ public function generateLevel($name, $seed = null, $options = []) {
 
 		$seed = $seed === null ? (PHP_INT_SIZE === 8 ? unpack("N", @Utils::getRandomBytes(4, false))[1] << 32 >> 32 : unpack("N", @Utils::getRandomBytes(4, false))[1]) : (int)$seed;
 
-		$provider = LevelProviderManager::getProviderByName($providerName = "mcregion");
+		$provider = LevelProviderManager::getProviderByName($providerName = "anvil");
```
Comment (JackNoordhuis) [inline src/pocketmine/Server.php]: Change to this so it doesn't try to load mcregion in anvil

`if(($provider = LevelProviderManager::getProviderByName($providerName = $this->getProperty("level-settings.default-format", "mcregion"))) === null) {
			$provider = LevelProviderManager::getProviderByName($providerName = "mcregion");
}`

File: src/pocketmine/level/format/anvil/ChunkRequestTask.php [hunk 2]
```diff
@@ -32,6 +32,8 @@ class ChunkRequestTask extends AsyncTask {
 
 	protected $levelId;
 
+    protected $chunk;
+
```
Comment (JackNoordhuis) [inline src/pocketmine/level/format/anvil/ChunkRequestTask.php]: Formatting

File: src/pocketmine/level/format/anvil/ChunkRequestTask.php [hunk 3]
```diff
@@ -57,7 +59,8 @@ public function __construct(Anvil $level, $levelId, $chunkX, $chunkZ) {
 		if(!($chunk instanceof Chunk)) {
 			throw new ChunkException("Invalid Chunk sent");
 		}
-		$this->biomeIds = $chunk->getBiomeIdArray();
+        $this->chunk = $chunk;
+        $this->biomeIds = $chunk->getBiomeIdArray();
```
Comment (JackNoordhuis) [inline src/pocketmine/level/format/anvil/ChunkRequestTask.php]: More formatting errors

File: src/pocketmine/level/format/anvil/ChunkRequestTask.php [hunk 4]
```diff
@@ -103,7 +106,7 @@ public function onRun() {
 				$orderedLight .= $this->getHalfColumn($blockLight, $x, $z);
 			}
 		}
-
+        $chunk = Chunk::fromFastBinary($this->chunk);
```
Comment (JackNoordhuis) [inline src/pocketmine/level/format/anvil/ChunkRequestTask.php]: Formatting again

File: src/pocketmine/Server.php [hunk 5]
```diff
@@ -884,9 +884,7 @@ public function generateLevel($name, $seed = null, $options = []) {
 
 		$seed = $seed === null ? (PHP_INT_SIZE === 8 ? unpack("N", @Utils::getRandomBytes(4, false))[1] << 32 >> 32 : unpack("N", @Utils::getRandomBytes(4, false))[1]) : (int)$seed;
 
-		if(($provider = LevelProviderManager::getProviderByName($providerName = $this->getProperty("level-settings.default-format", "mcregion"))) === null) {
-			$provider = LevelProviderManager::getProviderByName($providerName = "anvil");
-		}
+		if(($provider = LevelProviderManager::getProviderByName($providerName = $this->getProperty("level-settings.default-format", "mcregion"))) === null) {$provider = LevelProviderManager::getProviderByName($providerName = "anvil");}
```
Comment (JackNoordhuis) [inline src/pocketmine/Server.php]: wtf is this

**Comment by IrishPacks** [comment]
I'll make a new PR.

**Pull request closed without merging**